File: tools/WagsMediaGenerator/Models/Link.cs
namespace WagsMediaGenerator.Models;

public class Link
{
    public int LinkId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Url { get; set; } = string.Empty;
    
    public string Author { get; set; } = string.Empty;
    
    public DateTime LinkDate { get; set; }
    
    public int ReadingLogIssueNumber { get; set; }

    public Tag Type { get; set; } = new();

    public Tag Category { get; set; } = new();
}

File: api/src/models/link.ts
export type LinkCategory = {
    linkCategoryId: number;
    name: string;
    colorCode: string;
    linkCount?: number;
}

export type LinkType = {
    linkTypeId: number;
    name: string;
    colorCode: string;
}

export type Link = {
    linkId: number;
    linkTypeId: number;
    linkCategoryId: number;
    title: string;
    url: string;
    author: string;
    linkDate: Date | string | null;
    readingLogIssueNumber: number;
    category?: LinkCategory;
    type?: LinkType;
}

export type LinkQueryReturn = {
    LinkId: number;
    LinkTypeId: number;
    Title: string;
    Url: string;
    Author: string;
    LinkDate: Date;
    ReadingLogIssueNumber: number;
    LinkCategoryId: number;
    LinkCategoryName: string;
    LinkCategoryColor: string;
    LinkTypeName: string;
    LinkTypeColor: string;
}

export type LinkCategoryQueryReturn = {
    LinkCategoryId: number;
    Name: string;
    ColorCode: string;
    LinkCount: number;
}
